overlord/servicestate/service_control_test.go

```diff
@@ -773,6 +782,66 @@ func (s *serviceControlSuite) TestRestartServices(c *C) {
 	})
 }
 
+func (s *serviceControlSuite) TestRestartWithExplicitServices(c *C) {
```

Comment:
a variant of this test where Services == ExplicitServices would be another good test case to have I think